src/Libraries/Microsoft.Extensions.DataIngestion.Markdig/MarkdownParser.cs

```diff
@@ -0,0 +1,279 @@
+            }
+
+            rootSection.Elements.Add(MapBlock(documentMarkdown, previousWasBreak, block));
+            previousWasBreak = false;
+        }
+
+        return result;
+    }
+
+    private static bool IsEmptyBlock(Block block) // Block with no text. Sample: QuoteBlock the next block is a quote.
+        => block is LeafBlock emptyLeafBlock && (emptyLeafBlock.Inline is null || emptyLeafBlock.Inline.FirstChild is null);
+
+    private static IngestionDocumentElement MapBlock(string documentMarkdown, bool previousWasBreak, Block block)
+    {
+        string elementMarkdown = documentMarkdown.Substring(block.Span.Start, block.Span.Length);
+
+        IngestionDocumentElement element = block switch
+        {
+            LeafBlock leafBlock => MapLeafBlockToElement(leafBlock, previousWasBreak, elementMarkdown),
+            ListBlock listBlock => MapListBlock(listBlock, previousWasBreak, documentMarkdown, elementMarkdown),
+            QuoteBlock quoteBlock => MapQuoteBlock(quoteBlock, previousWasBreak, documentMarkdown, elementMarkdown),
+            Table table => new IngestionDocumentTable(elementMarkdown, GetCells(table, documentMarkdown)),
+            _ => throw new NotSupportedException($"Block type '{block.GetType().Name}' is not supported.")
+        };
+
+        return element;
+    }
+
+    private static IngestionDocumentElement MapLeafBlockToElement(LeafBlock block, bool previousWasBreak, string elementMarkdown)
+        => block switch
+        {
+            HeadingBlock heading => new IngestionDocumentHeader(elementMarkdown)
+            {
+                Text = GetText(heading.Inline),
+                Level = heading.Level
+            },
+            ParagraphBlock footer when previousWasBreak => new IngestionDocumentFooter(elementMarkdown)
+            {
+                Text = GetText(footer.Inline),
+            },
+            ParagraphBlock image when image.Inline!.Descendants<LinkInline>().FirstOrDefault() is LinkInline link && link.IsImage => MapImage(elementMarkdown, link),
+            ParagraphBlock paragraph => new IngestionDocumentParagraph(elementMarkdown)
+            {
+                Text = GetText(paragraph.Inline),
+            },
+            CodeBlock codeBlock => new IngestionDocumentParagraph(elementMarkdown)
+            {
+                Text = GetText(codeBlock.Inline),
+            },
+            _ => throw new NotSupportedException($"Block type '{block.GetType().Name}' is not supported.")
+        };
+
+    private static IngestionDocumentImage MapImage(string elementMarkdown, LinkInline link)
+    {
+        IngestionDocumentImage result = new(elementMarkdown);
+
+        // ![Alt text](data:image/png;base64,...)
+        if (link.FirstChild is LiteralInline literal)
+        {
+            result.AlternativeText = literal.Content.ToString();
+        }
+
+        if (link.Url is not null && link.Url.StartsWith("data:image/png;base64,", StringComparison.Ordinal))
```

Comment:
What about other reasonably possible formats? Like jpeg, gif or webp, from top of my head?